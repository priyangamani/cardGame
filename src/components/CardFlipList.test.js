
import CardFlipList from './cardFlipContainer/CardFlipList';
import React from "react"
import { generatePairNumberList,CHOOSE_NUMBER_OF_CARD_PAIRS } from '../utils/GeneratePairsNumList';
import { cleanup, fireEvent, render, waitFor } from "@testing-library/react-native"

describe("<CardFlipList />", () => {

    afterEach(() => {
        cleanup()
      })

    let component
    let mockCardItemList =  jest.fn()
    beforeEach(() => {
        component = (
         <CardFlipList listItems={generatePairNumberList(CHOOSE_NUMBER_OF_CARD_PAIRS)} restart={mockCardItemList} />
        )
      })

  
    test("renders correctly", () => {
      // Act
      const tree = render(component)
      // Assert
      expect(tree).toMatchSnapshot()
    })
  
    test("content components exists", () => {
      const { getByTestId, container } = render(component)
      expect(container).toBeDefined()
      expect(() => getByTestId("view-card-flip-container")).toBeDefined()
      expect(() => getByTestId("step-count-text")).toBeDefined()
      expect(() => getByTestId("restart-button")).toBeDefined()
    })

    test("restart Button click", () => {
        const { getByTestId } = render(component)
        const restartButton = getByTestId("restart-button")
        expect(restartButton).toBeDefined()
        fireEvent.press(restartButton)
      })
      test("flatList container", () => {
        const { getByTestId, debug } = render(component)
        const flatListContainer = getByTestId("flatlist-container")
        expect(flatListContainer).toBeDefined()
        expect(getByTestId("flatlist-container")).not.toBeNull()
        expect(flatListContainer.props.renderItem).toBeDefined()
      })
  })
  