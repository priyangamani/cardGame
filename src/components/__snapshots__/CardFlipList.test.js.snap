// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFlipList /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 40,
      "paddingTop": 20,
    }
  }
  testID="view-card-flip-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      testID="step-count-text"
    >
      Step Count: 
      0
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="restart-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Restart
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 10,
      }
    }
    data={
      Array [
        Object {
          "show": false,
          "value": 53,
        },
        Object {
          "show": false,
          "value": 28,
        },
        Object {
          "show": false,
          "value": 66,
        },
        Object {
          "show": false,
          "value": 66,
        },
        Object {
          "show": false,
          "value": 53,
        },
        Object {
          "show": false,
          "value": 98,
        },
        Object {
          "show": false,
          "value": 10,
        },
        Object {
          "show": false,
          "value": 98,
        },
        Object {
          "show": false,
          "value": 10,
        },
        Object {
          "show": false,
          "value": 22,
        },
        Object {
          "show": false,
          "value": 28,
        },
        Object {
          "show": false,
          "value": 22,
        },
      ]
    }
    disableVirtualization={false}
    extraData={
      Array [
        Object {
          "show": false,
          "value": 53,
        },
        Object {
          "show": false,
          "value": 28,
        },
        Object {
          "show": false,
          "value": 66,
        },
        Object {
          "show": false,
          "value": 66,
        },
        Object {
          "show": false,
          "value": 53,
        },
        Object {
          "show": false,
          "value": 98,
        },
        Object {
          "show": false,
          "value": 10,
        },
        Object {
          "show": false,
          "value": 98,
        },
        Object {
          "show": false,
          "value": 10,
        },
        Object {
          "show": false,
          "value": 22,
        },
        Object {
          "show": false,
          "value": 28,
        },
        Object {
          "show": false,
          "value": 22,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="flatlist-container"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 5,
              }
            }
            testID="view-card-flip-item"
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 150,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                    ],
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "marginBottom": "auto",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "marginTop": "auto",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
